backend/services/implementations/schoolService.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  ISchoolService,
+  SchoolRequestDTO,
+  SchoolResponseDTO,
+} from "../interfaces/schoolService";
+import MgSchool, { School } from "../../models/school.model";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+import { UserDTO } from "../../types";
+import IUserService from "../interfaces/userService";
+
+const Logger = logger(__filename);
+
+class SchoolService implements ISchoolService {
+  userService: IUserService;
+
+  constructor(userService: IUserService) {
+    this.userService = userService;
+  }
+
+  /**
+   * This method creates a new school in the database.
+   *
+   * @param school The request object containing information about the school
+   * to create
+   */
+  async createSchool(school: SchoolRequestDTO): Promise<SchoolResponseDTO> {
+    let teacherDTOs: Array<UserDTO>;
+    let newSchool: School | null;
+    try {
+      // get the user details for each teacher
+      teacherDTOs = await this.userService.findAllUsersByIds(school.teachers);
+
+      // check that all teachers exist as users
+      if (teacherDTOs.length !== school.teachers.length) {
+        throw new Error("One or more of the teacher IDs was not found");
+      }
+
+      // create a new School document
+      newSchool = await MgSchool.create({ ...school });
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to create school. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+
+    return {
+      id: newSchool.id,
+      name: newSchool.name,
+      country: newSchool.country,
+      subRegion: newSchool.subRegion,
+      city: newSchool.city,
+      address: newSchool.address,
+      teachers: teacherDTOs,
```

Comment:
quick question.. i was wondering why we are returning `teacherDTOs` vs just the teacher ids?

Reply:
This is so that we can get access to the teacher objects when we call this method - otherwise, we would have to make a separate call to get the teacher information. Either way works - just thought it'd save us some work later on to query this information.

Reply:
makes sense, ty!!